app/controllers/publish/courses/a_level_requirements/a_level_requirements_controller.rb

```diff
@@ -33,12 +34,15 @@ def create
 
         private
 
+        def add_flash_message; end
+
         def verify_teacher_degree_apprenticeship_course
           redirect_to publish_provider_recruitment_cycle_courses_path(provider_code: provider.provider_code, recruitment_cycle_year: provider.recruitment_cycle_year) unless @course.teacher_degree_apprenticeship?
         end
 
         def assign_course
-          @course = CourseDecorator.new(provider.courses.find_by!(course_code: params[:course_code]))
+          @course = provider.courses.find_by!(course_code: params[:course_code])
+          @course_decorator = CourseDecorator.new(@course)
```

Comment:
I have to make the distinction between the active record model and the decorator. The decorator is used in the views the active model passing to the wizard more above in this class